src/store/modules/user/actions.ts

```diff
@@ -222,6 +225,102 @@ const actions: ActionTree<UserState, RootState> = {
       console.error(error);
     }
     return []
+  },
+
+  async setProductIdentificationPref({ commit, state }, payload) {
+    let prefValue = JSON.parse(JSON.stringify(state.productIdentificationPref))
+
+    // when selecting none as ecom store, not updating the pref as it's not possible to save pref with empty productStoreId
+    if(!(state.currentEComStore as any).productStoreId) {
+      commit(types.USER_PREF_PRODUCT_IDENT_CHANGED, prefValue)
+      return;
+    }
+
+    prefValue[payload.id] = payload.value
+
+    const params = {
+      "fromDate": prefValue.fromDate,
+      "productStoreId": (state.currentEComStore as any).productStoreId,
+      "settingTypeEnumId": "PRODUCT_STORE_PREF",
+      "settingValue": JSON.stringify({'primaryId': prefValue.primaryId,'secondaryId': prefValue.secondaryId})
+    }
+
+    try {
+      const resp = await UserService.updateProductStoreSetting(params) as any
+
+      if(resp.status == 200) {
+        showToast(translate('Product identifier preference updated'))
+      } else {
+        showToast(translate('Failed to update product identifier preference'))
+        prefValue = JSON.parse(JSON.stringify(state.productIdentificationPref))
+      }
+    } catch(err) {
+      showToast(translate('Failed to update product identifier preference'))
+      prefValue = JSON.parse(JSON.stringify(state.productIdentificationPref))
+      console.error(err)
+    }
+    commit(types.USER_PREF_PRODUCT_IDENT_CHANGED, prefValue)
+  },
+
+  async createProductIdentificationPref({ commit, state }, payload) {
+    const params = {
+      "fromDate": payload.fromDate,
+      "productStoreId": (state.currentEComStore as any).productStoreId,
+      "settingTypeEnumId": "PRODUCT_STORE_PREF",
+      "settingValue": JSON.stringify({'primaryId': payload.primaryId,'secondaryId': payload.secondaryId})
+    }
+
+    try {
+      const resp = await UserService.createProductStoreSetting(params) as any
+
+      if(resp.status == 200) {
+        commit(types.USER_PREF_PRODUCT_IDENT_CHANGED, payload)
+      } else {
+        commit(types.USER_PREF_PRODUCT_IDENT_CHANGED, JSON.parse(JSON.stringify(state.productIdentificationPref)))
```

Comment:
I think this might not be needed, we will be doing create operation for setting the initial value which is already there in state

Reply:
Sir, we need to define the else part in case when the eComStore selected does not have a pref set and we got an error in create api call.

Reply:
Updated Sir